### PR TITLE
Renaming of datadog agent and packaging

### PR DESCRIPTION
**Pull request by craffit**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1569,7 +1569,7 @@ the Datadog user will need to be modified from ```'datadog'@'localhost'``` to ``
 * If you were using a custom check that needed python dependencies you will have to reinstall them using the bundled pip:
 
        ```
-sudo /opt/datadog-agent/embedded/bin/pip install YOUR_DEPENDENCY
+sudo /opt/stackstate-agent/embedded/bin/pip install YOUR_DEPENDENCY
        ```
 * Configuring checks in datadog.conf for checks.d is deprecated and won't work anymore. Please configure your checks by editing the yaml files in the conf.d directory.
```
Comment (ktatarnikov) [inline CHANGELOG.md:1574]: datadog.conf. please check if CHANGELOG.md goes to package

File: README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-[![Build Status](https://travis-ci.org/DataDog/dd-agent.svg?branch=master)](https://travis-ci.org/DataDog/dd-agent)
+[![Build Status](https://travis-ci.org/StackVista/sts-agent.svg?branch=manual_renames)](https://travis-ci.org/StackVista/sts-agent)
 
 The Datadog Agent faithfully collects events and metrics and brings
```
Comment (ktatarnikov) [inline README.md:3]: Datadog in readme

File: checks.d/go_expvar.py [hunk 3]
```diff
@@ -153,7 +148,7 @@ def parse_expvar_data(self, data, tags, metrics, max_metrics, namespace):
 
                 if count >= max_metrics:
                     self.warning("Reporting more metrics than the allowed maximum. "
-                                 "Please contact support@datadoghq.com for more information.")
+                                 "Please contact support@stackstate.com for more information.")
```
Comment (ktatarnikov) [inline checks.d/go_expvar.py]: is it support@stackstate.com? or info@stackstate.com? please just double check our support url

File: conf.d/docker.yaml.example [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # Warning
-# The user running the Datadog Agent (usually "dd-agent") must be part of the "docker" group
+# The user running the StackState Agent (usually "dd-agent") must be part of the "docker" group
```
Comment (ktatarnikov) [inline conf.d/docker.yaml.example]: dd-agent reference

File: config.py [hunk 5]
```diff
@@ -441,7 +420,7 @@ def get_config(parse_args=True, cfg_path=None, options=None):
         elif get_os() == 'windows':
             # default windows location
             common_path = _windows_commondata_path()
-            agentConfig['additional_checksd'] = os.path.join(common_path, 'Datadog', 'checks.d')
+            agentConfig['additional_checksd'] = os.path.join(common_path, 'StackState', 'checks.d')
 
         if config.has_option('Main', 'use_dogstatsd'):
             agentConfig['use_dogstatsd'] = config.get('Main', 'use_dogstatsd').lower() in ("yes", "true")
```
Comment (ktatarnikov) [inline config.py:426]: we might have "use_stsstatsd" in configuration...

File: packaging/datadog-agent/source/install_agent.sh [hunk 6]
```diff
@@ -1,9 +1,4 @@
 #!/bin/bash
-# (C) Datadog, Inc. 2010-2016
-# All rights reserved
-# Licensed under Simplified BSD License (see LICENSE)
-# Datadog Agent installation script: install and set up the Agent on supported Linux distributions
-# using the package manager and Datadog repositories.
 
 set -e
 logfile="ddagent-install.log"
```
Comment (ktatarnikov) [inline packaging/datadog-agent/source/install_agent.sh:4]: ddagent-install.log?

File: packaging/osx/install.sh [hunk 7]
```diff
@@ -1,9 +1,5 @@
 #!/bin/bash
-# (C) Datadog, Inc. 2010-2016
-# All rights reserved
-# Licensed under Simplified BSD License (see LICENSE)
 
-# Datadog Agent install script for Mac OS X.
 set -e
 logfile=ddagent-install.log
```
Comment (ktatarnikov) [inline packaging/osx/install.sh:4]: ddagent-install.log

File: packaging/osx/install.sh [hunk 8]
```diff
@@ -1,9 +1,5 @@
...
 dmg_file=/tmp/datadog-agent.dmg
```
Comment (ktatarnikov) [inline packaging/osx/install.sh:5]: datadog-agent.dmg

File: packaging/osx/install.sh [hunk 9]
```diff
@@ -29,11 +25,7 @@ function on_error() {
     printf "\033[31m$ERROR_MESSAGE
 It looks like you hit an issue when trying to install the Agent.
 
-Troubleshooting and basic usage information for the Agent are available at:
-
-    http://docs.datadoghq.com/guides/basic_agent_usage/
-
-If you're still having problems, please send an email to support@datadoghq.com
+Please send an email to info@stackstate.com
 with the contents of ddagent-install.log and we'll do our very best to help you
```
Comment (ktatarnikov) [inline packaging/osx/install.sh:29]: ddagent-install.log

File: packaging/osx/install.sh [hunk 10]
```diff
@@ -109,15 +98,15 @@ done
...
-If you ever want to stop the Agent, please use the Datadog Agent App or
-datadog-agent command.
+If you ever want to stop the Agent, please use the Stackstate Agent App or
+stackstate-agent command.
 
 It will start automatically at login, if you want to enable it at startup,
 run these commands: (the agent will still run as your user)
 
-    sudo cp '/opt/datadog-agent/etc/com.datadoghq.agent.plist' /Library/LaunchDaemons
+    sudo cp '/opt/stackstate-agent/etc/com.datadoghq.agent.plist' /Library/LaunchDaemons
```
Comment (ktatarnikov) [inline packaging/osx/install.sh:109]: datadoghq

File: stackstate.conf.example [hunk 11]
```diff
@@ -142,25 +137,23 @@ gce_updated_hostname: yes
...
-# DogStatsd configuration                                                    #
-# DogStatsd is a small server that aggregates your custom app metrics. For   #
-# usage information, check out http://docs.datadoghq.com/guides/dogstatsd/   #
+# StsStatsd configuration                                                    #
+# StsStatsd is a small server that aggregates your custom app metrics. For   #
 # ========================================================================== #
 
-# If you don't want to enable the DogStatsd server, set this option to no
+# If you don't want to enable the StsStatsd server, set this option to no
 # use_dogstatsd: yes
```
Comment (ktatarnikov) [inline stackstate.conf.example:145]: this is the property use_dogstatsd in config I reffered previously...

File: tests/core/test_config.py [hunk 12]
```diff
@@ -194,7 +194,7 @@ def test_conf_path_to_check_name(self, *args):
         """
         # Samples
         check_name = u"haproxy"
-        unix_check_path = u"/etc/dd-agent/conf.d/haproxy.yaml"
+        unix_check_path = u"/etc/sts-agent/conf.d/haproxy.yaml"
         win_check_path = u"C:\\ProgramData\\Datadog\\conf.d\\haproxy.yaml"
```
Comment (ktatarnikov) [inline tests/core/test_config.py:198]: Windows check path refer to Datadog

**Review by ktatarnikov — CHANGES_REQUESTED** [review]
please check my comments...

File: tests/core/test_config.py (hunk 12 above)
Comment (raboof) [reply tests/core/test_config.py:198]: I think we can leave windows and osx out of scope for now

File: stackstate.conf.example (hunk 11 above)
Comment (raboof) [reply stackstate.conf.example:145]: let's discuss with @markbakker78 if we really want to rename dogstatsd and how far we want to take this

File: packaging/osx/install.sh (hunk 10 above)
Comment (raboof) [reply packaging/osx/install.sh:109]: I think we can leave osx out of scope for now

File: CHANGELOG.md (hunk 1 above)
Comment (raboof) [reply CHANGELOG.md:1574]: @craffit suggests CHANGELOG doesn't make it to the package - I'll remove this change

File: README.md (hunk 2 above)
Comment (raboof) [reply README.md:3]: I think we don't ship the README, either

File: conf.d/docker.yaml.example (hunk 4 above)
Comment (raboof) [reply conf.d/docker.yaml.example]: Good one, there were still a lot of these - added them to `./sts-check-naming.sh` and fixed them

File: packaging/datadog-agent/source/install_agent.sh (hunk 6 above)
Comment (raboof) [reply packaging/datadog-agent/source/install_agent.sh:4]: i think we can keep installing from source out of scope for now

**Comment by raboof** [comment]
Open problems:
- We removed licensing headers, I don't think we should want that. Let's check
- We have incompletely renamed dogstatsd. I think we should consider if we want to move forward with that

**Pull request merged**